### PR TITLE
Adds a "Toggle" circuit component

### PR DESCRIPTION
**Pull request by TheSmallBlue**
## About The Pull Request
This circuit component receives a signal, and when it does it toggles between outputting a 0 or a 1.

## Why It's Good For The Game
While this is already possible to do with current circuit components, I personally feel like this component would help newcomers to circuitry by reserving the RAM component for actual value storage and retrieval. 
The current way of making a toggleable input consists of three components minimum: the signal input, a RAM component and a comparison component, as seen below with this "flashlight" circuit of sorts.
4dCDJAl2Kg
From experience I can say that the RAM component is a bit daunting for people who have never done circuits before. While it is understood that it can be used for saving values, you'd think that it'd be use in more complex circuits, not one that makes a simple function such as a toggle.
The toggle component helps on this by simplifying it all to just two components, the signal input and the toggle component, as seen below with the same flashlight circuit.
image

TL;DR: Makes it less daunting to do a really basic thing.

## Changelog
:cl:
expansion: Added the Toggle circuit component
/:cl:

### CODE REVIEW

**Comment by RaveRadbury** [comment]
You can use a NOT rather than a comparison.

**Comment by TheSmallBlue** [comment]
Same thing really, I just wanna simplify it to one component

**Comment by Watermelon914** [comment]
I don't really like this since it's overly specific and can already be achieved with current components. The point of components is to be as generic as possible and this just aims to make the RAM component more niche when it can already easily accomplish this task.

People new to circuitry will learn or be taught how to make simple toggles, because they only need to learn it once. Adding components that can already be made easily with current components is just a way for new players to skip learning how to create the logic for their own circuits.

**Comment by Watermelon914** [comment]
image
More basic example of a toggle component

**Comment by Mothblocks** [comment]
I'm not opposed to these simpler circuits, even if they're possible with other ocmponents--remember the following from the circuits design doc:

> The mechanics should be easy to pick up, and the UI should be easy to use. The challenge should come from optimization (power usage, evaluation time) and from thinking through the logistics of a design, the same challenges that come through traditional engineering.

If your concern is people not understanding fundamentals, remember that any complicated circuit is going to necessitate a clear understanding of how the mechanics work.

**Comment by TheBonded** [comment]
you can literally just hook up a NOT's output to it's own input to get the same result, still using one component

**Comment by TheSmallBlue** [comment]
Ok yeah this pull is kinda pointless, I'll close it. I do think that the RAM component needs a revamp to make it easier to understand for new people or that the whole "no looking at the wiki" thing in the design doc gets thrown out the window (which i think it should anyway)

**Comment by Watermelon914** [comment]
> I'm not opposed to these simpler circuits, even if they're possible with other components--remember the following from the circuits design doc:
> 
> > The mechanics should be easy to pick up, and the UI should be easy to use. The challenge should come from optimization (power usage, evaluation time) and from thinking through the logistics of a design, the same challenges that come through traditional engineering.
> 
> If your concern is people not understanding fundamentals, remember that any complicated circuit is going to necessitate a clear understanding of how the mechanics work.

It's more about people making simpler circuits for stuff that is already pretty simple and easy to do. If it's a more complex and complicated thing that can get messy with other circuits, then that's fine but basic logic like this being simplified just bypasses the need to think about stuff logically

**Comment by Watermelon914** [comment]
For this, I think the system would benefit greatly if the connections were a lot clearly if they're heading backwards because connections being behind components can look pretty ugly.

I want to look at improving the UI a lot more and introducing a lot more QoL to it so that making circuits can look even nicer.
